### PR TITLE
Active job aggregate metrics

### PR DESCRIPTION
**Pull request by mprimi**
This is a cherry-pick of https://github.com/Netflix/genie/pull/848 with significant changes from the original.

### CODE REVIEW

File: genie-docs/src/docs/asciidoc/_properties.adoc [hunk 1]
```diff
@@ -491,6 +491,16 @@ to the number of CPU cores x 2 + 1
 |1
 |no
 
+|genie.tasks.user-metrics.enabled
+|Whether or not to publish user-tagged metrics
+|true
+|no
+
+|genie.tasks.user-metrics.period
```
Comment (mprimi) [inline genie-docs/src/docs/asciidoc/_properties.adoc]: Bug: this is actually `refreshInterval` in the properties class

File: genie-web/src/main/java/com/netflix/genie/web/jpa/services/JpaServiceUtils.java [hunk 2]
```diff
@@ -301,4 +303,17 @@ static JobMetadata toJobMetadataDto(final JobMetadataProjection jobMetadataProje
             entity.setMetadata(null);
         }
     }
+
+    static UserResourcesSummary toUserResourceSummaryDto(
+        final UserJobResourcesAggregate userJobResourcesAggregate
+    ) {
+        final String user = userJobResourcesAggregate.getUser();
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/jpa/services/JpaServiceUtils.java]: As an alternative you could just have the `getters` in the interface return `Optional` and use `orElse()`

File: genie-web/src/main/java/com/netflix/genie/web/jpa/entities/aggregates/package-info.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/jpa/entities/aggregates/package-info.java]: year is off.

File: genie-web/src/main/java/com/netflix/genie/web/jpa/entities/aggregates/package-info.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+/**
+ * A package containing aggregates interfaces for Spring Data JPA.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@ParametersAreNonnullByDefault
+package com.netflix.genie.web.jpa.entities.aggregates;
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/jpa/entities/aggregates/package-info.java]: Shouldn't this be in the previous commit as you already had a class in there within this package?

File: genie-common/src/main/java/com/netflix/genie/common/dto/UserResourcesSummary.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * A summary of the resources used by a given user.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
```
[the remark this answers is not in the document]
Comment (mprimi) [reply genie-common/src/main/java/com/netflix/genie/common/dto/UserResourcesSummary.java]: Ok, i thought they'd be unnecessary in this particular case. But good to have them anyway.

File: genie-common/src/test/java/com/netflix/genie/common/dto/UserResourcesSummaryTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ */
+package com.netflix.genie.common.dto;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test UserResourcesSummary DTO.
+ */
+public class UserResourcesSummaryTest {
```
[the remark this answers is not in the document]
Comment (mprimi) [reply genie-common/src/test/java/com/netflix/genie/common/dto/UserResourcesSummaryTest.java]: Just went with what the rest of the package does.

File: genie-web/src/main/java/com/netflix/genie/web/jpa/entities/aggregates/package-info.java (hunk 4 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/jpa/entities/aggregates/package-info.java]: Commit are sorted by date in GH web UI, but this commit comes first.

**Pull request closed without merging**